src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -164,12 +165,38 @@ internal View TemplatedRoot
 					{
 						RegisterContentTemplateRoot();
 
-						OnApplyTemplate();
+						if (FeatureConfiguration.Control.UseDeferredOnApplyTemplate)
+						{
+							// It's too soon the call the ".OnApplyTemplate" method: it should be invoked after the "Loading" event.
+							_applyTemplateShouldBeInvoked = true;
+						}
+						else
+						{
+							OnApplyTemplate();
+						}
 					}
 				}
 			}
 		}
 
+		private bool _applyTemplateShouldBeInvoked = false;
+
+		private protected override void OnPostLoading()
+		{
+			base.OnPostLoading();
+
+			TryCallOnApplyTemplate();
+		}
+
+		private void TryCallOnApplyTemplate()
+		{
+			if (_applyTemplateShouldBeInvoked)
+			{
+				_applyTemplateShouldBeInvoked = false;
```

Comment:
If in the handler of the loading does `control.ApplyTemplate()`, the `OnApplyTemplate` will be invoked twice. You should reset the flag also in the `ApplyTemplate()`